src/core/jsproxy.c

```diff
@@ -2648,9 +2650,15 @@ wrap_promise(JsVal promise, JsVal done_callback, PyObject* js2py_converter)
   result = _PyObject_CallMethodIdNoArgs(loop, &PyId_create_future);
   FAIL_IF_NULL(result);
 
-  set_result = _PyObject_GetAttrId(result, &PyId_set_result);
+  helpers_mod = PyImport_ImportModule("_pyodide._future_helpers");
+  FAIL_IF_NULL(helpers_mod);
+  _Py_IDENTIFIER(get_future_resolvers);
+  helpers =
+    _PyObject_CallMethodIdOneArg(helpers, &PyId_get_future_resolvers, result);
```

Comment:
I guess this needs to be helpers_mod not helpers?

```suggestion
    _PyObject_CallMethodIdOneArg(helpers_mod, &PyId_get_future_resolvers, result);
```

Reply:
thanks!